pkg/collection/service.go

```diff
@@ -0,0 +1,25 @@
+package collection
+
+import (
+	"context"
+	"time"
+)
+
+type CollectFunction func(ctx context.Context, req *CollectRequest) error
+
+func (c CollectFunction) Collect(ctx context.Context, req *CollectRequest) error {
+	return c(ctx, req)
+}
+
+func NewChannelCollector(c chan *EventsBatch) Collector {
+	return CollectFunction(func(ctx context.Context, req *CollectRequest) error {
```

Comment:
Just a thought. I see the possibility of expanding this by including deserialization and metric collection (batch read metrics and events_rx_total). However, I haven't found a good way to describe what will this Collector do or what are the boundaries on this collector.

I'm thinking of this direction for the future extensibility of collector interface apart from HTTP, GRPC, and WS. So that, the future interfaces won't miss something like adding these metrics for example. Maybe this is too early to define it. We can wait for the delivery guarantee feature or another interface support before making further development on this layer.

Reply:
Agreed! let's wait for a few more releases and how the collector matures and then we can add more functionality.